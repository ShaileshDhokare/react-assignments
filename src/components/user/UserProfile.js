import React from 'react';
import PropTypes from 'prop-types';

const UserProfile = ({userDetails}) => {
    return (
        <div className="card my-4">
            <div className="card-header">
            <h5 className="card-title">
                User Profile
            </h5>
            </div>
            <div className="card-body">
                <p><b>Name:</b> {userDetails.name}</p>
                <p><b>UserName:</b> {userDetails.username}</p>
                <p><b>Email:</b> {userDetails.email}</p>
                <p><b>Phone:</b> {userDetails.phone}</p>
                <p><b>Website:</b> {userDetails.website}</p>
                <p><b>Company:</b> {userDetails.company.name}</p>
            </div>    
        </div>
    )
}

// UserProfile.propTypes = {
//     userDetails: PropTypes.object.isRequired
// };

export default UserProfile;
