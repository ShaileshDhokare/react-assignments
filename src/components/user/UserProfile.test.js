import Enzyme, { shallow} from 'enzyme';
import React from 'react';
import UserProfile from './UserProfile';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe('<UserProfile />', () => {
   it('Show user profile', () => {
      const userData = {
         name: 'test',
         username: 'test',
         email: 'test',
         phone: '12345',
         website: 'www.test.com',
         company: {
            name: 'Test Inc'
         }
      }
      const wrapper = shallow(<UserProfile userDetails={userData} />);
      expect(wrapper.props().userData.name).toEqual({name:'jest'});
      // expect(shallow(<UserProfile userDetails={userData} />)).toMatchSnapshot();
   })
})