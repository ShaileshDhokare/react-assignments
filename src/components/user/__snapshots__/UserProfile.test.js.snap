// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> Show user profile 1`] = `ShallowWrapper {}`;
